src/plugins/dist_matrixops/retile_annotations.cpp

```diff
@@ -0,0 +1,811 @@
+        execution_tree::localities_information&& arr_localities) const
+    {
+        using namespace execution_tree;
+
+        // current annotation information
+        std::uint32_t const loc_id = arr_localities.locality_.locality_id_;
+        std::uint32_t const num_localities =
+            arr_localities.locality_.num_localities_;
+        std::size_t dim = arr_localities.size();    // size of the whole array
+        tiling_information_1d tile_info(
+            arr_localities.tiles_[loc_id], name_, codename_);
+
+        std::int64_t cur_start = tile_info.span_.start_;
+        std::int64_t cur_stop = tile_info.span_.stop_;
+
+        // updating the annotation_ part of localities annotation
+        arr_localities.annotation_.name_ += "_retiled";
+        ++arr_localities.annotation_.generation_;
+
+        // is it a rwo vector or a column vector?
+        std::size_t span_index = 0;
+        if (!arr_localities.has_span(0))
+        {
+            HPX_ASSERT(arr_localities.has_span(1));
+            span_index = 1;
+        }
+
+        // desired annotation information
+        std::int64_t des_start, des_stop, des_size;
+        bool type_ = span_index;    // initialized by the current type
+
+        if (tiling_type == "user")
+        {
+
+            std::tie(type_, des_start, des_stop) =
+                detail::tile_extraction_1d(std::move(new_tiling));
+            des_size = des_stop - des_start;
+            if (des_size <= 0)
+            {
+                HPX_THROW_EXCEPTION(hpx::bad_parameter,
+                    "dist_matrixops::primitives::retile_annotations::retile1d",
+                    generate_error_message(
+                        "the given start point of the new_tiling should be less "
+                        "than its stop point"));
+            }
+        }
+        else    // tiling_type is one of "sym", "row" or "column"
+        {
+            if (tiling_type == "row")
+            {
+                type_ = true;
+            }
+            else if (tiling_type == "column")
+            {
+                type_ = false;
+            }
+            std::tie(des_start, des_size) =
+                tile_calculation::tile_calculation_1d(loc_id, dim, numtiles);
+
+            if (intersection != 0) // there should be some overlapped sections
+            {
+                std::tie(des_start, des_size) =
+                    tile_calculation::tile_calculation_overlap_1d(
+                        des_start, des_size, dim, intersection);
+            }
+            des_stop = des_start + des_size;
+        }
+
+
+        // creating the updated array as result
+        auto v = arr.vector();
+        blaze::DynamicVector<T> result(des_size);
+        util::distributed_vector<T> v_data(
+            arr_localities.annotation_.name_, v, num_localities, loc_id);
+
+        // relative start
+        std::int64_t rel_start = des_start - cur_start;
+        if (rel_start < 0 || des_stop > cur_stop)
```

Comment:
Should this be `des_stop >= cur_stop`?

Reply:
I think it should be `>`. If they are equal, still `else` part can get it. Am I missing something?

Reply:
Ok, I was merely asking as normally the upper limit is not part of a (open) range.